vtex-apps/b2b-organizations
# Move creation button to the topbar

#### What problem is this solving?

This is moving the button for creation organizations to the topbar, which is the new Admin standard for these page actions.

The change seems simple, but it required a lot more code than the previous and bigger visual change of my previous PR (related to updating layout + tabs). I didn't actually change much of the code though, I just refactored a big part of it so I could split the creation button + modal from the organizations list, since they were within the same file. I'll comment inline to further explain the changes.

#### How to test it?

Create a few organizations in the workspace below and check that they're being correctly created as before.

Also, if the new organization's name is such that it would show up at the top of the initial organizations table on the page, it correctly shows up there just like before. To test this, create a new organization that starts with the letter A for example, and check that it shows up on the table after the operation is successful.

[Workspace](https://mairamodal--b2bstoreqa.myvtex.com/admin/b2b-organizations/organizations/#/organizations)

#### Screenshots or example usage:

**Before**
<img width="1437" alt="Screenshot 2023-07-13 at 13 30 48" src="https://github.com/vtex-apps/b2b-organizations/assets/5216049/aa9fd201-5abf-4dd0-8ef4-98fd0b3760d4">

**After**
<img width="1440" alt="Screenshot 2023-07-14 at 11 56 16" src="https://github.com/vtex-apps/b2b-organizations/assets/5216049/687d15ee-fb5b-45bf-8f50-7c3defdd0df0">

#### How does this PR make you feel? [:link:](http://giphy.com/)

![](https://media.giphy.com/media/v1.Y2lkPTc5MGI3NjExbHdibG0xeHFyZTZwc2p1MWVwanR6ajBhOXF2ajB6azZsa3lyOXl4eCZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/l0HlGmv4WqldO9c5y/giphy.gif)
